### PR TITLE
chore: Refactor `createLayerSource()`

### PR DESCRIPTION
**Pull request by salisdemos**
split the method createLayerSource  into 3 methods. this should not affect anything. Layers (WMS and WMTS) should still work.

This is a side outcome while bugfixing [T34144](https://yaits.demos-deutschland.de/T34144)

### CODE REVIEW

File: client/js/components/map/publicdetail/Map.vue [hunk 1]
```diff
@@ -1587,6 +1539,60 @@ export default {
       return layers
     },
 
+    getWMTSSource (layer) {
+      const layerArray = Array.isArray(layer.attributes.layers) ? layer.attributes.layers : layer.attributes.layers.split(',')
+      const url = this.addGetCapabilityParamToUrl(layer.attributes.url)
+
+      return $.ajax({
```
Comment (gruenbergerdemos) [inline client/js/components/map/publicdetail/Map.vue:1546]: Why did you change this back to ajax?

Comment (salisdemos) [reply client/js/components/map/publicdetail/Map.vue:1546]: because ajax has this tiny "async: false" (what ever that does in real - because technically its always async) Option, which is nor provided by axios. And the code is so messy, that I wasn't able to refactor it, that it works asynchronously.

Comment (salisdemos) [reply client/js/components/map/publicdetail/Map.vue:1546]: this is a bit mixced up. I initially wanted to cherrypick https://github.com/demos-europe/demosplan-core/pull/1866. I thought the cherrypick-PR would be faster that this one. Then I waited for the approval for 1866 and created this PR in the meanwhile which lead to the mess...

**Review by gruenbergerdemos — APPROVED** [review]
lgtm
